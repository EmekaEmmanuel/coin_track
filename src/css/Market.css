* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: 'Montserrat', sans-serif;
  letter-spacing: 0.2px;
}

.app_container {
  margin: auto;
  background-color: #111315;
}

.market_main_tag {
  text-align: center;
}

.market_section1 {
  padding: 100px 0;
  width: 50%;
  margin: auto;
}

.market_font1 {
  font-size: 11px;
  color: #efefef9a;
}

.market_font2 {
  font-size: 50px;
  color: #efefef;
  padding: 20px 0;
}

.market_font3 {
  font-size: 11px;
  color: #efefef9a;
}

.market_section2 {
  width: 90%;
  margin: auto;
  padding: 40px 40px;
  background-color: #020a11;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  gap: 20px;
  border-radius: 15px;
}

.market_article {
  background-color: white;
  align-items: center;
  display: flex;
  justify-content: center;
  padding: 10px 5px;
  gap: 15px;
  color: #111315;
  border-radius: 15px;
  transition: 1s;
}

.market_article:hover {
  transform: scale(1.05);
}


.market_article_box {
  list-style: none;
  text-align: center;
}

.market_details_item {
  padding: 5px 0;
  color: #020a11;
}


@media only screen and (max-width: 1024px) {
  .market_section1 {
    padding: 100px 0;
    width: 60%;
    margin: auto;
  }

  .market_font2 {
    font-size: 45px;
    color: #efefef;
    padding: 20px 0;
  }

  .market_section2 {
    width: 90%;
    margin: auto;
    padding: 40px;
    background-color: #020a11;
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    gap: 20px;
    border-radius: 15px;
  }
}

@media only screen and (max-width: 768px) {
  .market_section1 {
    padding: 100px 0;
    width: 70%;
    margin: auto;
  }

  .market_font1 {
    font-size: 9px;
    color: #efefef9a;
  }

  .market_font2 {
    font-size: 40px;
    color: #efefef;
    padding: 20px 0;
  }

  .market_font3 {
    font-size: 10px;
    color: #efefef9a;
  }

  .market_section2 {
    width: 90%;
    margin: auto;
    padding: 40px;
    background-color: #020a11;
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    gap: 20px;
    border-radius: 15px;
  }
}

@media only screen and (max-width: 425px) {
  .market_section1 {
    padding: 100px 0;
    width: 90%;
    margin: auto;
  }

  .market_font2 {
    font-size: 28px;
    color: #efefef;
    padding: 20px 0;
  }

  .market_section2 {
    width: 90%;
    margin: auto;
    padding: 40px;
    background-color: #020a11;
    display: grid;
    grid-template-columns: repeat(1, 1fr);
    gap: 20px;
    border-radius: 15px;
  }
}

@media only screen and (max-width: 375px) {
  .market_font2 {
    font-size: 25px;
    color: #efefef;
    padding: 20px 0;
  }
}