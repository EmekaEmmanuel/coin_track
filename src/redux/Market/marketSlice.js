import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
  loading: false,
  markets: [],
  error: '',
};

export const fetchMarkets = createAsyncThunk('market/fetchMarkets', async (id) => {
  const { data } = await (axios.get(`https://api.coinstats.app/public/v1/markets?coinId=${id}`));
  return data;
});

/* eslint-disable no-param-reassign */

const marketSlice = createSlice({
  name: 'market',
  initialState,
  extraReducers: (builder) => {
    builder.addCase(fetchMarkets.pending, (state) => {
      state.loading = true;
    })
      .addCase(fetchMarkets.fulfilled, (state, action) => {
        state.loading = false;
        state.error = '';
        state.markets = action.payload;
      })
      .addCase(fetchMarkets.rejected, (state) => {
        state.loading = false;
        state.error = '';
      });
  },
});

/* eslint-disable no-param-reassign */

export default marketSlice.reducer;
