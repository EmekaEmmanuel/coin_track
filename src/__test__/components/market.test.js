import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../redux/store';
import Market from '../../Components/Market';

it('Market component renders correctly', () => {
  const myMarkets = store.getState().market.markets;
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <Market myMarkets={myMarkets} />
      </MemoryRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
