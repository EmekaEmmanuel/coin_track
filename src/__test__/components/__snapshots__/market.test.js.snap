// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Market component renders correctly 1`] = `
<div
  className="app_container"
>
  <main
    className="market_main_tag"
  >
    <section
      className="market_section1"
    >
      <p
        className="market_font1"
      >
        BE THE FIRST TO KNOW ABOUT MARKET CHANGES
      </p>
      <h1
        className="market_font2"
      >
        Get an Overview on multi market exchanges
      </h1>
      <p
        className="market_font3"
      >
        Market multiplicity enables users to manage and trade their assets with greater transparency in a permissionless environment with a simplified and straightforward interface.
      </p>
    </section>
    <section
      className="market_section2"
    />
  </main>
</div>
`;
